_posts/2018-08-30-on-the-fly-transformations.md

```diff
@@ -0,0 +1,342 @@
+from periodic boundary conditions, which cause some issues with some molecular
+viewers (PyMol for example), removing the rotation and translation of a particular
+molecule and/or centering it in the unit cell, which helps focus on the its actual
+conformational changes by removing their natural movement in solution.
+These transformations help us better identifying patterns in the behavior of our
+biological systems, and, more importantly, showing them to the world.
+
+## The advantage of using MDAnalysis for trajectory transformations
+Many simulation packages often contain tools to transform and analyze trajectories,
+such as GROMACS' `trjconv`. However, most of the times, the user is required to apply
+all the intended transformations to the whole trajectory (or the portion of interest)
+prior to visualization and analysis. This often requires processing huge files, sometimes
+more than once. Moreover, some tools such as `trjconv` do not support frame indexing
+for the most popular trajectory formats, requiring iterating over frames that are not
+needed for that particular analysis.
+Trajectory transformations in MDAnalysis, on the other end, have one great advantage - they
+are performed on-the-fly for each frame that is read. Transformations are added to a universe
+as a transformation workflow containing one or more transformations. The API also makes it easy
+to add new transformations for your own projects.
+Another things that really makes the "on-the-fly" aspect of the MDAnalysis transformations
+shine is coupling it to a visualization widget such as
+[NGL Viewer](http://nglviewer.org/ngl/api/index.html).
+
+## Using MDAnalysis transformations
+Now it's time to learn how to use the trajectory transformations in MDAnalysis. During the
+following steps, we will apply some transformations on a 1 ns trajectory of a simple
+19-residue peptide embeded in a 128-DMPC membrane, showing the GROMACS `trjconv` command
+and the equivalent MDAnalysis code and output. To keep thinks lightweight, frames are
+were taken every 100 ps, and water molecules were removed. This can be easily done with
+MDAnalysis.
+
+### Example 1: making everything whole again
+When performing MD simulations using periodic boundary conditions, molecules will often
+cross the limits of the unit cell. When this happens, some atoms of the molecule will
+show up on the the opposing side of the unit cell and some molecular viewers will show
+stretched bonds and other visual artifacts depending on the visual representation of the
+system. 
+This is the case of our system. Without any modifications, when we look at the trajectory
+of our system, things become more cluttered and confusing:
+
+
+```python
+import warnings
+warnings.filterwarnings('ignore') # some attributes are missing 
+import MDAnalysis as mda
+import nglview as nv
+u = mda.Universe('pept_in_memb.tpr', 'pept_in_memb.xtc')
+nv.show_mdanalysis(u)
```

Comment:
@lilyminium would there be a way to embed the nglview widget here with actual content?